src/core/server/audit_trail/types.ts

```diff
@@ -19,17 +19,78 @@
 import { KibanaRequest } from '../http';
 
 /**
- * Event to audit.
+ * Audit event using ECS format: https://www.elastic.co/guide/en/ecs/1.5/index.html
  * @public
  *
  * @remarks
  * Not a complete interface.
  */
-export interface AuditableEvent {
+export interface AuditEvent {
   message: string;
-  type: string;
+  event: {
+    action: string;
+    category?: EventCategory | EventCategory[];
+    type?: EventType | EventType[];
+    outcome?: EventOutcome;
+    module?: string;
+    dataset?: string;
+  };
+  user: {
+    name: string;
+    email?: string;
+    full_name?: string;
+    hash?: string;
+  };
+  kibana?: {
+    session_id?: string;
+    user_roles?: string[];
+    saved_objects?: Array<{
+      id?: string;
+      type: string;
+      namespaces?: string[];
+    }>;
+  };
```

Comment:
Maybe change to something like this?
```suggestion
  kibana?: {
    session_id?: string;
    user_roles?: string[];
    namespace: string;
    saved_object_crud?: Array<{
      id: string;
      type: string;
    }>;
    saved_object_share?: {
      id: string;
      type: string;
      namespaces: {
        before: string[];
        after: string[];
      }
    };
  };
```

Note 1: the `namespace` would be 'default' instead of undefined (so it can be easily compared to the contents of `namespaces` below).
Note 2: the required `id` is an assumption that we can always get this info from the `create` operation (see https://github.com/elastic/kibana/pull/74640#discussion_r469310730; TBD, I need to investigate more fully).